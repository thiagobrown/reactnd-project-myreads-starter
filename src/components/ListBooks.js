import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import BookShelf from './BookShelf'
import Loader from './Loader'
import { shelfs } from './../utils/constants'

const ListBooks = ({ books = [], loading = false, onUpdateBook }) => {

    return (
        <div className="list-books">
            <div className="list-books-title">
                <h1>MyReads</h1>
            </div>
            <div className="list-books-content">
                {
                    loading
                        ? <Loader />
                        : (
                            <div>
                                {shelfs.map(s => (
                                    <BookShelf key={s.shelf} title={s.title}
                                        books={books.filter(book => book.shelf === s.shelf)}
                                        onUpdateBook={onUpdateBook} />
                                ))}
                            </div>
                        )
                }
            </div>
            <div className="open-search">
                <Link to="/search" />
            </div>
        </div>
    )
}

ListBooks.propTypes = {
    books: PropTypes.array,
    loading: PropTypes.bool,
    onUpdateBook: PropTypes.func.isRequired
}

export default ListBooks;