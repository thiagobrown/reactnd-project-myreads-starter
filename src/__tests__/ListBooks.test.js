import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import ListBooks from './../components/ListBooks'

describe('Testing ListBooks Component', () => {
  let props = {
    books: [{
      id: 1,
      title: 'Book One',
      authors: ['Authors One'],
      imageLinks: { thumbnail: 'url/book/one' },
      shelf: 'read'
    }, {
      id: 2,
      title: 'Book Two',
      authors: ['Authors Two'],
      shelf: 'wantToRead'
    }, {
      id: 3,
      title: 'Book Three',
      imageLinks: { thumbnail: 'url/book/three' },
      shelf: 'currentlyReading'
    }],
    loading: false,
    onUpdateBook: jest.fn()
  }

  const wrapper = shallow(<ListBooks {...props} />);

  it('render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});