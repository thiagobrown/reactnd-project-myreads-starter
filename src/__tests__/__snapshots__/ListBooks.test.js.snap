exports[`Testing ListBooks Component render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListBooks
    books={
        Array [
            Object {
              "authors": Array [
                "Authors One",
              ],
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "url/book/one",
              },
              "shelf": "read",
              "title": "Book One",
            },
            Object {
              "authors": Array [
                "Authors Two",
              ],
              "id": 2,
              "shelf": "wantToRead",
              "title": "Book Two",
            },
            Object {
              "id": 3,
              "imageLinks": Object {
                "thumbnail": "url/book/three",
              },
              "shelf": "currentlyReading",
              "title": "Book Three",
            },
          ]
    }
    loading={false}
    onUpdateBook={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="list-books-title">
          <h1>
                    MyReads
          </h1>
</div>,
        <div
          className="list-books-content">
          <div>
                    <BookShelf
                              books={
                                        Array [
                                                  Object {
                                                    "id": 3,
                                                    "imageLinks": Object {
                                                      "thumbnail": "url/book/three",
                                                    },
                                                    "shelf": "currentlyReading",
                                                    "title": "Book Three",
                                                  },
                                                ]
                              }
                              onUpdateBook={[Function]}
                              title="Currently Reading" />
                    <BookShelf
                              books={
                                        Array [
                                                  Object {
                                                    "authors": Array [
                                                      "Authors Two",
                                                    ],
                                                    "id": 2,
                                                    "shelf": "wantToRead",
                                                    "title": "Book Two",
                                                  },
                                                ]
                              }
                              onUpdateBook={[Function]}
                              title="Want to Read" />
                    <BookShelf
                              books={
                                        Array [
                                                  Object {
                                                    "authors": Array [
                                                      "Authors One",
                                                    ],
                                                    "id": 1,
                                                    "imageLinks": Object {
                                                      "thumbnail": "url/book/one",
                                                    },
                                                    "shelf": "read",
                                                    "title": "Book One",
                                                  },
                                                ]
                              }
                              onUpdateBook={[Function]}
                              title="Read" />
          </div>
</div>,
        <div
          className="open-search">
          <Link
                    replace={false}
                    to="/search" />
</div>,
      ],
      "className": "list-books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            MyReads
</h1>,
          "className": "list-books-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MyReads",
          },
          "ref": null,
          "rendered": "MyReads",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <BookShelf
                        books={
                                    Array [
                                                Object {
                                                  "id": 3,
                                                  "imageLinks": Object {
                                                    "thumbnail": "url/book/three",
                                                  },
                                                  "shelf": "currentlyReading",
                                                  "title": "Book Three",
                                                },
                                              ]
                        }
                        onUpdateBook={[Function]}
                        title="Currently Reading" />
            <BookShelf
                        books={
                                    Array [
                                                Object {
                                                  "authors": Array [
                                                    "Authors Two",
                                                  ],
                                                  "id": 2,
                                                  "shelf": "wantToRead",
                                                  "title": "Book Two",
                                                },
                                              ]
                        }
                        onUpdateBook={[Function]}
                        title="Want to Read" />
            <BookShelf
                        books={
                                    Array [
                                                Object {
                                                  "authors": Array [
                                                    "Authors One",
                                                  ],
                                                  "id": 1,
                                                  "imageLinks": Object {
                                                    "thumbnail": "url/book/one",
                                                  },
                                                  "shelf": "read",
                                                  "title": "Book One",
                                                },
                                              ]
                        }
                        onUpdateBook={[Function]}
                        title="Read" />
</div>,
          "className": "list-books-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BookShelf
                books={
                                Array [
                                                Object {
                                                  "id": 3,
                                                  "imageLinks": Object {
                                                    "thumbnail": "url/book/three",
                                                  },
                                                  "shelf": "currentlyReading",
                                                  "title": "Book Three",
                                                },
                                              ]
                }
                onUpdateBook={[Function]}
                title="Currently Reading" />,
              <BookShelf
                books={
                                Array [
                                                Object {
                                                  "authors": Array [
                                                    "Authors Two",
                                                  ],
                                                  "id": 2,
                                                  "shelf": "wantToRead",
                                                  "title": "Book Two",
                                                },
                                              ]
                }
                onUpdateBook={[Function]}
                title="Want to Read" />,
              <BookShelf
                books={
                                Array [
                                                Object {
                                                  "authors": Array [
                                                    "Authors One",
                                                  ],
                                                  "id": 1,
                                                  "imageLinks": Object {
                                                    "thumbnail": "url/book/one",
                                                  },
                                                  "shelf": "read",
                                                  "title": "Book One",
                                                },
                                              ]
                }
                onUpdateBook={[Function]}
                title="Read" />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "currentlyReading",
              "nodeType": "function",
              "props": Object {
                "books": Array [
                  Object {
                    "id": 3,
                    "imageLinks": Object {
                      "thumbnail": "url/book/three",
                    },
                    "shelf": "currentlyReading",
                    "title": "Book Three",
                  },
                ],
                "onUpdateBook": [Function],
                "title": "Currently Reading",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "wantToRead",
              "nodeType": "function",
              "props": Object {
                "books": Array [
                  Object {
                    "authors": Array [
                      "Authors Two",
                    ],
                    "id": 2,
                    "shelf": "wantToRead",
                    "title": "Book Two",
                  },
                ],
                "onUpdateBook": [Function],
                "title": "Want to Read",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "read",
              "nodeType": "function",
              "props": Object {
                "books": Array [
                  Object {
                    "authors": Array [
                      "Authors One",
                    ],
                    "id": 1,
                    "imageLinks": Object {
                      "thumbnail": "url/book/one",
                    },
                    "shelf": "read",
                    "title": "Book One",
                  },
                ],
                "onUpdateBook": [Function],
                "title": "Read",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/search" />,
          "className": "open-search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "replace": false,
            "to": "/search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="list-books-title">
            <h1>
                        MyReads
            </h1>
</div>,
          <div
            className="list-books-content">
            <div>
                        <BookShelf
                                    books={
                                                Array [
                                                            Object {
                                                              "id": 3,
                                                              "imageLinks": Object {
                                                                "thumbnail": "url/book/three",
                                                              },
                                                              "shelf": "currentlyReading",
                                                              "title": "Book Three",
                                                            },
                                                          ]
                                    }
                                    onUpdateBook={[Function]}
                                    title="Currently Reading" />
                        <BookShelf
                                    books={
                                                Array [
                                                            Object {
                                                              "authors": Array [
                                                                "Authors Two",
                                                              ],
                                                              "id": 2,
                                                              "shelf": "wantToRead",
                                                              "title": "Book Two",
                                                            },
                                                          ]
                                    }
                                    onUpdateBook={[Function]}
                                    title="Want to Read" />
                        <BookShelf
                                    books={
                                                Array [
                                                            Object {
                                                              "authors": Array [
                                                                "Authors One",
                                                              ],
                                                              "id": 1,
                                                              "imageLinks": Object {
                                                                "thumbnail": "url/book/one",
                                                              },
                                                              "shelf": "read",
                                                              "title": "Book One",
                                                            },
                                                          ]
                                    }
                                    onUpdateBook={[Function]}
                                    title="Read" />
            </div>
</div>,
          <div
            className="open-search">
            <Link
                        replace={false}
                        to="/search" />
</div>,
        ],
        "className": "list-books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              MyReads
</h1>,
            "className": "list-books-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MyReads",
            },
            "ref": null,
            "rendered": "MyReads",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <BookShelf
                            books={
                                          Array [
                                                        Object {
                                                          "id": 3,
                                                          "imageLinks": Object {
                                                            "thumbnail": "url/book/three",
                                                          },
                                                          "shelf": "currentlyReading",
                                                          "title": "Book Three",
                                                        },
                                                      ]
                            }
                            onUpdateBook={[Function]}
                            title="Currently Reading" />
              <BookShelf
                            books={
                                          Array [
                                                        Object {
                                                          "authors": Array [
                                                            "Authors Two",
                                                          ],
                                                          "id": 2,
                                                          "shelf": "wantToRead",
                                                          "title": "Book Two",
                                                        },
                                                      ]
                            }
                            onUpdateBook={[Function]}
                            title="Want to Read" />
              <BookShelf
                            books={
                                          Array [
                                                        Object {
                                                          "authors": Array [
                                                            "Authors One",
                                                          ],
                                                          "id": 1,
                                                          "imageLinks": Object {
                                                            "thumbnail": "url/book/one",
                                                          },
                                                          "shelf": "read",
                                                          "title": "Book One",
                                                        },
                                                      ]
                            }
                            onUpdateBook={[Function]}
                            title="Read" />
</div>,
            "className": "list-books-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <BookShelf
                  books={
                                    Array [
                                                      Object {
                                                        "id": 3,
                                                        "imageLinks": Object {
                                                          "thumbnail": "url/book/three",
                                                        },
                                                        "shelf": "currentlyReading",
                                                        "title": "Book Three",
                                                      },
                                                    ]
                  }
                  onUpdateBook={[Function]}
                  title="Currently Reading" />,
                <BookShelf
                  books={
                                    Array [
                                                      Object {
                                                        "authors": Array [
                                                          "Authors Two",
                                                        ],
                                                        "id": 2,
                                                        "shelf": "wantToRead",
                                                        "title": "Book Two",
                                                      },
                                                    ]
                  }
                  onUpdateBook={[Function]}
                  title="Want to Read" />,
                <BookShelf
                  books={
                                    Array [
                                                      Object {
                                                        "authors": Array [
                                                          "Authors One",
                                                        ],
                                                        "id": 1,
                                                        "imageLinks": Object {
                                                          "thumbnail": "url/book/one",
                                                        },
                                                        "shelf": "read",
                                                        "title": "Book One",
                                                      },
                                                    ]
                  }
                  onUpdateBook={[Function]}
                  title="Read" />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "currentlyReading",
                "nodeType": "function",
                "props": Object {
                  "books": Array [
                    Object {
                      "id": 3,
                      "imageLinks": Object {
                        "thumbnail": "url/book/three",
                      },
                      "shelf": "currentlyReading",
                      "title": "Book Three",
                    },
                  ],
                  "onUpdateBook": [Function],
                  "title": "Currently Reading",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "wantToRead",
                "nodeType": "function",
                "props": Object {
                  "books": Array [
                    Object {
                      "authors": Array [
                        "Authors Two",
                      ],
                      "id": 2,
                      "shelf": "wantToRead",
                      "title": "Book Two",
                    },
                  ],
                  "onUpdateBook": [Function],
                  "title": "Want to Read",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "read",
                "nodeType": "function",
                "props": Object {
                  "books": Array [
                    Object {
                      "authors": Array [
                        "Authors One",
                      ],
                      "id": 1,
                      "imageLinks": Object {
                        "thumbnail": "url/book/one",
                      },
                      "shelf": "read",
                      "title": "Book One",
                    },
                  ],
                  "onUpdateBook": [Function],
                  "title": "Read",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/search" />,
            "className": "open-search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "replace": false,
              "to": "/search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
